tests/integration/test_source_git_update_source_git.py

```diff
@@ -0,0 +1,122 @@
+# Copyright Contributors to the Packit project.
+# SPDX-License-Identifier: MIT
+
+import pytest
+
+from packit.exceptions import PackitException
+from packit.patches import PatchMetadata
+from packit.constants import DISTRO_DIR
+
+
+@pytest.fixture
+def update_api(api_instance_source_git):
+    # The version in dg is different from up, sync it
+    version = api_instance_source_git.up.specfile.get_version()
+    api_instance_source_git.dg.specfile.set_version(version)
+    api_instance_source_git.dg.specfile.save()
+    api_instance_source_git.dg.commit("Update spec", "")
+    return api_instance_source_git
+
+
+def test_update_source_git_version_differs(
+    sourcegit_and_remote,
+    distgit_and_remote,
+    update_api,
+):
+    """Checks that an error is thrown when version between dist-git and
+    source-git repos differ."""
+    sourcegit, _ = sourcegit_and_remote
+    distgit, _ = distgit_and_remote
+    update_api.up.specfile.set_version("1.2.3.4")
+    with pytest.raises(PackitException):
+        update_api.update_source_git("HEAD")
+
+
+def test_update_source_git_sources_changed(
+    sourcegit_and_remote,
+    distgit_and_remote,
+    update_api,
+):
+    """Checks that an error is thrown when sources in dist-git were modified."""
+    sourcegit, _ = sourcegit_and_remote
+    distgit, _ = distgit_and_remote
+    with open(distgit / "sources", "w") as sources:
+        sources.write("abcd")
```

Comment:
```suggestion
    (distgit / "sources").write_text("abcd")
```

There are more places like this in the file